firmware/quadruna/VC/src/hw/hw_adc.c

```diff
@@ -0,0 +1,27 @@
+#include "hw_adc.h"
+#include <assert.h>
+#include "hw_hal.h"
+#include "hw_adcConversions.h"
+#include "hw_utils.h"
+
+static uint16_t raw_adc_values[NUM_ADC_CHANNELS];
+static float    adc_voltages[NUM_ADC_CHANNELS];
+
+void HAL_ADC_ConvCpltCallback(ADC_HandleTypeDef *hadc)
+{
+    for (AdcChannel i = ADC1_CHANNEL_10; i < NUM_ADC_CHANNELS; i++)
```

Comment:
this should be "i = ADC1_CHANNEL_3" not "i = ADC1_CHANNEL_10" based on the order you added them in hw_adc.h enum